enhancements/machine-api/aws-placement-groups.md

```diff
@@ -0,0 +1,487 @@
+  // It is recommended to only use this value if multiple MachineSets share
+  // a single Placement Group, in which case, each MachineSet should represent an individual partition number.
+  // If unset, when a Partition placement group is used, AWS will attempt to
+  // distribute instances evenly between partitions.
+  // +kubebuilder:validation:Minimum:=1
+  // +kubebuilder:validation:Maximum:=7
+  // +optional
+  PartitionNumber int32 `json:"partitionNumber,omitempty"`
+}
+
+// AWSPlacementGroupType represents the valid values for the Placement GroupType field.
+type AWSPlacementGroupType string
+
+const (
+  // AWSClusterPlacementGroupType is the "Cluster" placement group type.
+  // Cluster placement groups place instances close together to improve network latency and throughput.
+  AWSClusterPlacementGroupType   AWSPlacementGroupType = "Cluster"
+  // AWSPartitionPlacementGroupType is the "Partition" placement group type.
+  // Partition placement groups reduce the likelihood of hardware failures
+  // disrupting your application's availability.
+  // Partition placement groups are recommended for use with large scale
+  // distributed and replicated workloads.
+  AWSPartitionPlacementGroupType AWSPlacementGroupType = "Partition"
+  // AWSSpreadPlacementGroupType is the "Spread" placement group type.
+  // Spread placement groups place instances on distinct racks within the availability
+  // zone. This ensures instances each have their own networking and power source
+  // for maximum hardware fault tolerance.
+  // Spread placement groups are recommended for a small number of critical instances
+  // which must be kept separate from one another.
+  // Using a Spread placement group imposes a limit of seven instances within
+  // the placement group within a single availability zone.
+  AWSSpreadPlacementGroupType    AWSPlacementGroupType = "Spread"
+)
+```
+
+### Implementation Details/Notes/Constraints
+
+#### Placement Group Strategies
+
+AWS offers three different placement group strategies.
+Each strategy is designed with a different use case in mind and each has it's own caveats.
+The details of each are fleshed out below.
+
+##### Cluster Placement Groups
+
+A Cluster placement group is a grouping of instances within a single Availability zone.
+Cluster placement groups are used for workloads that require higher per-flow throughput limits for TCP/IP traffic and
+as such, are placed in the same high-bisection bandwidth segment of the network.
+
+Cluster placement groups are recommended for applications that benefit from low network latency, high network
+throughput, or both.
+In particular, we expect that these will be used in conjunction with EFA network interfaces in HPC applications.
+
+When using Cluster placement groups, AWS recommends to launch all instances as the same instance type and if
+possible, to launch instances simultaneously to prevent the likelihood that capacity limits are reached within the
```

Comment:
if it becomes impossible to scale a machine up in the same placementgroup, how will the error be presented to the user?

Reply:
The Machine will enter a `Failed` phase, and the error will be presented to them on the status of the Machine itself.

If they want to resolve the issue, they can either wait until a later point (at which point more capacity may be available), or attempt to launch an new set of instances by scaling the MachineSet to 0, and back up, or just creating a new MachineSet with a new placement group and then removing the old one.

Having spoken to some contacts at AWS, to reach this limit, the user normally would need to be launching 1000s of instances in the same placement group. They launched over a 1000 c5.24xlarge in the same placement group without issue. This is all detailed in the placement group capacity risk. 

Due to the stats provided by AWS, we're pretty content that this is a low risk